sympy/printing/latex.py

```diff
@@ -3049,6 +3079,15 @@ def latex(expr, **settings):
     max: Integer or None, optional
         Sets the upper bound for the exponent to print floating point numbers in
         fixed-point format.
+    mat_compact: boolean, optional
+        Truncate large Matrix for printing. Default is ``True``, to avoid printing
+        slowdown.
+    mat_visible_rows: int, optional
+        Sets the maximum number of rows to be printed. Default is ``15``, and
+        it doesn't take effect if mat_compact is ``False``.
+    mat_visible_cols: int, optional
+        Sets the maximum number of columns to be printed. Default is ``15``, and
+        it doesn't take effect if mat_compact is ``False``.
```

Comment:
I don't think that this many matrix specific options are warranted.

Let's start with something basic: we can have an option like `truncate` that defaults to true. It could be used for more than just matrices but for now it could be implemented only for matrices. Ideally other printers like `str/repr` should support the same and it should be controllable via `init_printing`.

Reply:
Ok, I just made the changes for latex printing.  I agree truncation should also work for the other printers and I will try to get that working.  Should I put that in this PR or open a new one?

Also, do you think there is a way to implement this for many of the printers simultaneously(srepr, unicode, etc), or will it have to be a new implementation for each individual printer?